test/parallel/test-runner-run-coverage.mjs

```diff
@@ -0,0 +1,202 @@
+      const stream = run({ files, coverage: true, coverageExcludeGlobs: ['test/*/test-runner/invalid-tap.js'] });
+      stream.on('test:fail', common.mustNotCall());
+      stream.on('test:pass', common.mustCall(1));
+      stream.on('test:coverage', common.mustCall(({ summary: { files } }) => {
+        const filesPaths = files.map(({ path }) => path);
+        assert.strictEqual(filesPaths.some((path) => path.includes('test-runner/invalid-tap.js')), false);
+      }));
+      // eslint-disable-next-line no-unused-vars
+      for await (const _ of stream);
+    });
+
+    await it('should run with coverage and include by glob', async () => {
+      const stream = run({
+        files,
+        coverage: true,
+        coverageIncludeGlobs: ['test/fixtures/test-runner/coverage.js', 'test/*/v8-coverage/throw.js'],
+      });
+      stream.on('test:fail', common.mustNotCall());
+      stream.on('test:pass', common.mustCall(1));
+      stream.on('test:coverage', common.mustCall(({ summary: { files } }) => {
+        const filesPaths = files.map(({ path }) => path);
+        assert.strictEqual(filesPaths.some((path) => path.includes('v8-coverage/throw.js')), true);
+      }));
+      // eslint-disable-next-line no-unused-vars
+      for await (const _ of stream);
+    });
+
+    await it('should run with and include and excluded by globs', async () => {
+      const stream = run({
+        files,
+        coverage: true,
+        coverageIncludeGlobs: ['test/fixtures/test-runner/*.js'],
+        coverageExcludeGlobs: ['test/fixtures/test-runner/*-tap.js']
+      });
+      stream.on('test:fail', common.mustNotCall());
+      stream.on('test:pass', common.mustCall(1));
+      stream.on('test:coverage', common.mustCall(({ summary: { files } }) => {
+        const filesPaths = files.map(({ path }) => path);
+        assert.strictEqual(filesPaths.every((path) => !path.includes('test-runner/invalid-tap.js')), true);
+        assert.strictEqual(filesPaths.some((path) => path.includes('test-runner/coverage.js')), true);
+      }));
+      // eslint-disable-next-line no-unused-vars
+      for await (const _ of stream);
+    });
+
+    await it('should run with coverage and fail when below line threshold', async () => {
+      const thresholdErrors = [];
+      const stream = run({ files, coverage: true, lineCoverage: 99, branchCoverage: 99, functionCoverage: 99 });
+      stream.on('test:fail', common.mustNotCall());
+      stream.on('test:pass', common.mustCall(1));
+      stream.on('test:diagnostic', common.mustCall(({ message }) => {
+        const match = message.match(/Error: \d{2}\.\d{2}% (line|branch|function) coverage does not meet threshold of 99%/);
+        if (match) {
+          thresholdErrors.push(match[1]);
+        }
+      }));
+      // eslint-disable-next-line no-unused-vars
+      for await (const _ of stream);
+      assert.deepStrictEqual(thresholdErrors.sort(), ['branch', 'function', 'line']);
+    });
```

Comment:
As it is currently setup, the `process.exitCode` will be set to `1` when coverage fails. However, in run mode, this behavior must be changed, as it causes the file to exit early and fail.

Reply:
#54813 will fix this.